pytorch_lightning/trainer/states.py

```diff
@@ -17,9 +17,10 @@
 from typing import Callable, Optional
 
 import pytorch_lightning
+from pytorch_lightning.utilities import LightningEnum
 
 
-class TrainerState(str, Enum):
+class TrainerState(LightningEnum):
```

Comment:
In the docs, it says:

```
    >>> # which is case sensitive
    >>> TrainerState.FINISHED == 'finished'
    False
```

which is wrong with `LightningEnum`. Can you update it?